oxide_core/quaternion.rs

```diff
@@ -0,0 +1,169 @@
+use crate::vector::vector3::*;
+use crate::vector::*;
+use std::ops;
+
+#[derive(Copy, Clone, PartialEq, Debug)]
+pub struct Quaternion {
+  pub x: f32,
+  pub y: f32,
+  pub z: f32,
+  pub w: f32,
+}
+
+pub trait VectorPart {
+  fn get_vector_part(&self) -> Vector3;
+}
```

Comment:
Is there a reason why VectorPart should be a trait?

Reply:
Oh, I'll remove it and use just ```impl``` without making traits.

Reply:
✅ [`02d4435` (#14)](https://github.com/utilForever/Oxide/pull/14/commits/02d4435b02e47cc1c162a13c47ef5aead1cdcbc4)